aten/src/ATen/TensorImpl.cpp

```diff
@@ -85,4 +103,10 @@ void * TensorImpl::unsafeGetTH(bool retain) {
   return tensor;
 }
 
+std::unique_ptr<Storage> TensorImpl::storage() {
+  StorageImpl* storage = tensor->storage_;
```

Comment:
won't this crash on sparse tensors?

Reply:
Not sure. All tests are passing. @ezyang ?

Reply:
The method is [overwritten for sparse tensors](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/SparseTensorImpl.cpp#L51).

Reply:
Yeah, I don't think we urgently need to devirtualize this function (though we should eventually).